portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,13 +56,21 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("comment");
-    PreparedQuery results = datastore.prepare(query);
-    ArrayList<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String)entity.getProperty("value"));
+    String maxCommentsString = request.getParameter("max_comments");
+    if (maxCommentsString != null) {
+      Integer maxComments = Integer.parseInt(maxCommentsString);
+      Query query = new Query(COMMENT_TAG);
+      PreparedQuery results = datastore.prepare(query);
+      ArrayList<String> comments = new ArrayList<>();
+      int i = 0;
+      for (Entity entity : results.asIterable()) {
+        if (i++ == maxComments) break;
+        comments.add((String)entity.getProperty("value"));
+      }
```

Comment:
Is there a potentially more efficient method of retrieving N comments available through the PreparedQuery APIs?

Reply:
Addressed